### ISSUE TITLE
integration tests freeze with golang 1.10

### ISSUE THREAD

**Issue by Roasbeef** [lightningnetwork/lnd#768]
I recently upgraded to golang 1.10. I enjoyed the test caching, and swifter compilation times right up until I attempted to run the integration tests (with the proper build flags of course):
```
⛰ go test -v -logoutput -tags rpctest
=== RUN   TestLightningNetworkDaemon
```

Rather than proceeding to execute all our test cases, I observed that it just hung there (possibly indefinitely). Downgrading to golang 1.9.4 seems to fix the issue, so it's either the case that there was some regression in golang 1.10, or my installation was messed up somehow?

**Comment by halseth** [comment]
Same for me.

**Comment by Roasbeef** [comment]
Dug into a bit more, and it appears that the _first_ RPC command to `btcd` is the thing that fails. So this isn't _directly_ related to `lnd`. Will poke around a bit more to see what I can find.

**Comment by halseth** [comment]
Reinstalling `btcd` with go 1.10 did not help.

**Comment by Roasbeef** [comment]
Fixed by 39f9ae7cace142a2310a6a50f7ef7366f73b4198.
